lib/template/tests/test_template.c

```diff
@@ -126,6 +132,7 @@ setup(void)
 void
 teardown(void)
 {
+  scratch_buffers_explicit_gc();
```

Comment:
The scratch buffer is only used in the thread `format_template_thread`, do we really have to call gc here?

Reply:
The scratch buffers system is also used in other tests, but in those threads app_startup() covers the initialization.

With this change scratch-buffers are used in all template function invocations, be them either multi or single threaded.